torch/distributed/distributed_c10d.py

```diff
@@ -753,10 +755,35 @@ def recv(tensor,
         return src
 
 
+def async_collective(func):
```

Comment:
If this isn't meant to be used directly by PT users, maybe prefix it with an `_`?